packages/material-ui/src/Menu/Menu.js

```diff
@@ -44,23 +50,50 @@ const Menu = React.forwardRef(function Menu(props, ref) {
     disableAutoFocusItem = false,
     MenuListProps = {},
     onClose,
+    onEnter,
     onEntering,
+    onEntered,
     open,
     PaperProps = {},
+    setParentLastEnteredItemIndex,
```

Comment:
This name is very descriptive but can we shorten it at all? What about if we remove the "Entered": setParentLastItemIndex?